cmd/controller/main.go

```diff
@@ -70,3 +86,16 @@ func main() {
 		WithWebhooks(ctx, webhooks.NewWebhooks()...).
 		Start(ctx)
 }
+
+func validateK8sVersion(ctx context.Context, op *operator.Operator) error {
+	serverVersion := lo.Must1(op.VersionProvider.Get(ctx))
+	floatServerVersion := lo.Must1(strconv.ParseFloat(serverVersion, 32))
+
+	// We will only error if the user is running karpenter on a k8s version,
+	// that is out of the range of the minK8sVersion and maxK8sVersion
+	if minK8sVersion > floatServerVersion && floatServerVersion < maxK8sVersion {
```

Comment:
```suggestion
	if minK8sVersion > floatServerVersion || floatServerVersion < maxK8sVersion {
```

I think this logic is incorrect. You want to validate that _either_ the current version is less than the minK8sVersion _or_ that the current version is greater than the `maxK8sVersion`. It's `||`, not `&&`